pkg/cloudsources/client.go

```diff
@@ -0,0 +1,20 @@
+package cloudsources
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/cloudsources/discoveredclusters"
+	"github.com/stackrox/rox/pkg/cloudsources/paladin"
+)
+
+// Client exposes functionality to fetch discovered cluster from cloud sources.
+type Client interface {
+	GetDiscoveredClusters(ctx context.Context) ([]*discoveredclusters.DiscoveredCluster, error)
+}
+
+// NewClientForCloudSource creates a new Client based on the cloud source to fetch discovered clusters.
+func NewClientForCloudSource(source *storage.CloudSource) Client {
+	// For the time being, this only supports paladin cloud clients.
+	return paladin.NewClient(source)
```

Comment:
We should check the type here or make sure we filter for paladin cloud in the `ListCloudSources` call.